posts/views.py

```diff
@@ -32,24 +29,32 @@ def post(self, request):
             instance.skey =get_random_string(length=9)
             instance.save()
             if not request.user.is_authenticated:
-                url_path = reverse('posts:edit', kwargs={'id': instance.id , 'skey' : instance.skey})
+                url_path = reverse('posts:anon_edit', kwargs={'id': instance.id , 'skey' : instance.skey})
                 messages.success(request, mark_safe("<a href='{url_path}'>{url_path}</a>".format(url_path=url_path)))
             return redirect('posts:post', id=instance.id)
         return render(request, 'posts/create_post.html', {'form': form})
 
 class EditPost(View):
 
-    def get(self, request ,id , skey):
-        post = get_object_or_404(Post, id= id, skey= skey)
+    def dispatch(self, request, *args, **kwargs):
+        self.blog = get_object_or_404(Post, id=kwargs.get('id'))
+        if not self.blog.creator:
+            self.blog = get_object_or_404(Post, id=kwargs.get('id'),skey=kwargs.get('skey'))
```

Comment:
This function should be rewritten. Just check if its a skey link then filter either with `(id=id, skey=skey, creator=None)` or `(id=id, creator=request.user)`